plugins/cilium-cni/cni-uninstall.sh

```diff
@@ -30,6 +30,36 @@ HOST_PREFIX=${HOST_PREFIX:-/host}
 BIN_NAME=cilium-cni
 CNI_DIR=${CNI_DIR:-${HOST_PREFIX}/opt/cni}
 
+# Backwards compatibility
+if [ -n "${CILIUM_FLANNEL_MASTER_DEVICE}" ]; then
+	CILIUM_CNI_CHAINING_MODE="flannel"
+fi
+
+# cni-install.sh has a comparable case statement.
+# Make sure to keep both lists in sync.
+case "$CILIUM_CNI_CHAINING_MODE" in
+"flannel")
+	CNI_CONF_NAME=${CNI_CONF_NAME:-04-flannel-cilium-cni.conflist}
+	;;
+"generic-veth")
+	CNI_CONF_NAME=${CNI_CONF_NAME:-05-cilium.conflist}
+	;;
+"portmap")
+	CNI_CONF_NAME=${CNI_CONF_NAME:-05-cilium.conflist}
+	;;
+"aws-cni")
+	CNI_CONF_NAME=${CNI_CONF_NAME:-05-cilium.conflist}
+	;;
+*)
+	CNI_CONF_NAME=${CNI_CONF_NAME:-05-cilium.conf}
+	;;
+esac
+
+CILIUM_CNI_CONF=${CILIUM_CNI_CONF:-${HOST_PREFIX}/etc/cni/net.d/${CNI_CONF_NAME}}
+
+echo "Removing Cilium CNI configuration ${CILIUM_CNI_CONF}..."
+rm -f "${CILIUM_CNI_CONF}"
```

Comment:
Unfortunately this can't be accepted as it is. We don't have a way to tell if Cilium is being upgraded or uninstalled and therefore the configuration file needs to be kept to prevent Kubelet from potentially use another CNI plugin while Cilium is being upgraded. However, this could be done by the implementation of https://github.com/cilium/cilium/issues/13183 where the user is explicitly doing an uninstall.

Reply:
> For all container runtimes?

This seems to be the case for `kubelet` at least. Since this script is called from a `preStop` action in our k8s Pod specs, which other runtimes do we officially target and support?

Reply:
@ti-mo I meant that if kubelet picks another CNI plugin if the one available does not work.

Reply:
Why is integration with alternative runtimes an issue if the current approach doesn't even work with our primary target? (`kubelet` + `docker`)